tests/modules/images/webp-uploads/helper-tests.php

```diff
@@ -507,4 +507,27 @@ private function mock_empty_action( $action ) {
 		remove_all_actions( $action );
 		do_action( $action );
 	}
+
+	/**
+	 * Check different suffix values for generate_filename.
+	 *
+	 * @test
+	 */
+	public function it_should_check_different_suffix_values_for_generate_filename() {
+		$image_path = TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/car.jpeg';
+		$editor     = wp_get_image_editor( $image_path );
+		$editor->resize( 100, 50, true );
+		$ext     = pathinfo( $image_path, PATHINFO_EXTENSION );
+		$suffix  = $editor->get_suffix();
+		$suffix .= "-{$ext}";
+
+		// Test with no parameters.
+		$this->assertSame( 'car-100x50.jpeg', wp_basename( $editor->generate_filename() ) );
+		// Test with webp parameter.
+		$this->assertSame( 'car-100x50.webp', wp_basename( $editor->generate_filename( null, null, 'webp' ) ) );
+		// Test with suffix and webp parameter.
+		$this->assertSame( 'car-100x50-jpeg.webp', wp_basename( $editor->generate_filename( $suffix, null, 'webp' ) ) );
+		// Test with suffix and png parameter.
+		$this->assertSame( 'car-100x50-jpeg.png', wp_basename( $editor->generate_filename( $suffix, null, 'png' ) ) );
+	}
```

Comment:
This doesn't test the code change you have made above, instead it re-runs similar logic in a test.

I think a good test for this PR would be to call `webp_uploads_generate_additional_image_source()` for two different input files with similar file names, e.g. `image.jpeg` and `image.jpg`, and then to ensure that both images have different WebP files generated (`image-jpeg.webp` and `image-jpg.webp`).

Reply:
Thank you, Felix, for the review.

I have updated the PR with test coverage. Please re-review it and let me know in case any changes are needed.